app/graphql/concerns/validate_submission_gradable.rb

```diff
@@ -1,43 +1,59 @@
 module ValidateSubmissionGradable
   extend ActiveSupport::Concern
 
-  class OwnersShouldBeActive < GraphQL::Schema::Validator
+  class SubmissionShouldBeGradable < GraphQL::Schema::Validator
     def validate(_object, _context, value)
-      submission = TimelineEvent.find(value[:submission_id])
+      @error = nil
+      @submission = TimelineEvent.find_by(id: value[:submission_id])
 
-      #days since submission
+      submission_exists?(value[:submission_id]) && owners_active? &&
+        submission_live?
+      @error
```

Comment:
Let's avoid using `@error` by short-circuiting using `||` instead of `&&`:

```rb
@submission = TimelineEvent.find(value[:submission_id])

submission_must_exist(value[:submission_id]) || owners_must_be_active ||
  submission_must_be_live
```

In this case, each of the methods we call can either return `nil` or return the error text. For example:

```rb
def submission_must_be_live
  return unless @submission.archived?

  I18n.t("validate_submission_gradable.submission_should_be_live")
end
```